packages/semantic-ui/src/util.js

```diff
@@ -4,21 +4,69 @@ import React from "react";
  * Extract props meant for semantic UI components from props that are
  * passed to Widgets, Templates and Fields.
  * @param {Object} params
- * @param {Object} params.formContext
- * @param {Object} params.uiSchema
- * @param {Object} params.options
+ * @param {Object?} params.formContext
+ * @param {Object?} params.uiSchema
+ * @param {Object?} params.options
+ * @param {Object?} params.defaultSchemaProps
+ * @param {Object?} params.defaultContextProps
  * @returns {any}
  */
 export function getSemanticProps({
   formContext = {},
   uiSchema = {},
   options = {},
+  defaultSchemaProps = { fluid: true ,inverted: false },
+  defaultContextProps= { }
 }) {
+
+   const hasFormContextProps = formContext.semantic  ? true : false;
+   const hasSchemaProps = uiSchema["ui:options"] && uiSchema["ui:options"].semantic ? true : false;
+   const hasOptionsProps = options.semantic  ? true : false;
+   const formContextProps = hasFormContextProps ? formContext.semantic : defaultContextProps;
+   let schemaProps = hasSchemaProps ? uiSchema["ui:options"].semantic : defaultSchemaProps;
+   let optionProps = hasOptionsProps ? options.semantic : {};
+   // formContext props should overide other props
+   if (hasFormContextProps){
+     Object.keys(formContext.semantic).map((key) => {
```

Comment:
Do we need this code to get the formContext prop to override other props? Can't we remove this and afterwards just do:

```
  return Object.assign(
    {},
    schemaProps,
    optionProps,
    formContextProps,
  );
```